src/simpleppl.jl

```diff
@@ -0,0 +1,231 @@
+using AbstractPPL
+import Base.getindex
+using SparseArrays
+using Setfield
+using Setfield: PropertyLens, get
+
+"""
+    ModelState(value::NamedTuple{T}, input::NamedTuple{T}, eval::NamedTuple{T}, kind::NamedTuple{T})
+
+Record the state of the model as a struct of NamedTuples, all
+sharing the same key values, namely, those of the model parameters.
+`value` should store the initial/current value of the parameters.
+`input` stores a tuple of inputs for a given node. `eval` are the
+anonymous functions associated with each node. These might typically
+be either deterministic values or some distribution, but could an 
+arbitrary julia program. `kind` is a tuple of symbols indicating
+whether the node is a logical or stochastic node.
+"""
+struct ModelState{T}
```

Comment:
Maybe consider `DAGVarInfo` to stay consistent with the [`DynamicPPL` terminology ](https://github.com/TuringLang/DynamicPPL.jl/blob/1744ba7bbc9da5cd8e0d04ab66ba588030e30879/src/DynamicPPL.jl#L139)
```suggestion
struct GraphVarInfo{T} <: AbstractModelTrace
```

Reply:
@phipsgabler Since we have node attributes in `DAG`, we can store `value` and `input` in the same `values::NamedTuple`. If so, `GraphVarInfo` is basically the same as `SimpleVarInfo`. It motivates to perhaps moving `SimpleVarInfo` type definition into `AbstractPPL` in the future.  

https://github.com/TuringLang/DynamicPPL.jl/blob/master/src/simple_varinfo.jl#L129